app/src/main/java/com/example/abha_create_verify_android/utils/AadhaarXmlAndTextQr.kt

```diff
@@ -0,0 +1,148 @@
+                            AadhaarDataAttributes.AADHAAR_VTC_ATTR
+                        )
+
+                    scannedAadhaarCardInfo.subDistrict =
+                        parser.getAttributeValue(
+                            null,
+                            AadhaarDataAttributes.AADHAAR_SUB_DIST_ATTR
+                        )
+
+                    scannedAadhaarCardInfo.district =
+                        parser.getAttributeValue(
+                            null,
+                            AadhaarDataAttributes.AADHAAR_DIST_ATTR
+                        )
+
+                    scannedAadhaarCardInfo.state =
+                        parser.getAttributeValue(
+                            null,
+                            AadhaarDataAttributes.AADHAAR_STATE_ATTR
+                        )
+
+                    scannedAadhaarCardInfo.pinCode =
+                        parser.getAttributeValue(
+                            null,
+                            AadhaarDataAttributes.AADHAAR_PC_ATTR
+                        )
+
+                } else if (eventType == XmlPullParser.END_TAG) {
+                    Log.d("Aadhaar Xml", "End tag " + parser.name)
+                } else if (eventType == XmlPullParser.TEXT) {
+                    Log.d("Aadhaar Xml", "Text " + parser.text)
+                }
+                eventType = parser.next()
+            }
+
+            return
+        } catch (e: XmlPullParserException) {
+            Log.e("Exception", "Error in processing XML")
+            e.printStackTrace()
+            return
+        } catch (e: IOException) {
+            Log.e("Exception", e.toString())
+            e.printStackTrace()
+            return
+        }
+    }
+
+    private fun processPlainTextData(scanData: String?){
+        val keyValuePairs = scanData?.split(", ")
+        val result = mutableMapOf<String, String>()
+
+        if (keyValuePairs != null) {
+            for (pair in keyValuePairs) {
+                if(":" in pair) {
+                    val (key, value) = pair.split(":")
+                    result[key.trim().lowercase()] = value.trim()
+                }
+            }
+        }
+
+        scannedAadhaarCardInfo.name = result[AadhaarDataAttributes.AADHAAR_NAME_ATTR]
+        scannedAadhaarCardInfo.gender = result[AadhaarDataAttributes.AADHAAR_GENDER_ATTR]
+        scannedAadhaarCardInfo.dateOfBirth = result[AadhaarDataAttributes.AADHAAR_DOB_ATTR]
+        scannedAadhaarCardInfo.villageTownCity = result[AadhaarDataAttributes.AADHAAR_VTC_ATTR]
+        scannedAadhaarCardInfo.subDistrict = result[AadhaarDataAttributes.AADHAAR_SUB_DIST_ATTR]
+        scannedAadhaarCardInfo.district = result[AadhaarDataAttributes.AADHAAR_DIST_ATTR]
+        scannedAadhaarCardInfo.state = result[AadhaarDataAttributes.AADHAAR_STATE_ATTR]
+        scannedAadhaarCardInfo.pinCode = result[AadhaarDataAttributes.AADHAAR_PC_ATTR]
+    }
+
+    fun getAadhaarCardInfo(): AadhaarCardInfo {
```

Comment:
Is this function "getAadhaarCardInfo" getting used anywhere?

Reply:
not used now, but it is needed to read and display the output info

Reply:
if its gonna used in near future we can keep it.